packages/fiori-elements-writer/src/index.ts

```diff
@@ -82,11 +83,19 @@ async function generate<T extends {}>(basePath: string, data: FioriElementsApp<T
     await addOdataService(basePath, feApp.service, fs);
 
     const coercedUI5Version = semVer.coerce(feApp.ui5?.version)!;
+    // Determine if the changes preview should be enabled based on the project type and UI5 version
```

Comment:
Might be worth moving the templateOptions logic into a new function to keep the code clean.

Reply:
> Might be worth moving the templateOptions logic into a new function to keep the code clean and have a single condition for project type so its clear which options are for which project type

I moved this logic [here](https://github.com/SAP/open-ux-tools/pull/2132/files#diff-ff4edbeac8e42e9c7ce005e9ba864236b8e5909f154fde419beb25fa1098fa06R156)